### PR TITLE
C front-end: GCC >= 12 support __builtin_shufflevector

### PR DESCRIPTION
**Pull request by tautschnig**
This used to be Clang-only, but is now also supported by GCC.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/cbmc/gcc_vector3/main.c [hunk 1]
```diff
@@ -31,7 +31,12 @@ int main()
   assert(res.members[1] == 5);
   assert(res.members[2] == 3);
   assert(res.members[3] == 6);
-#elif defined(__clang__)
+#endif
+}
+
+void test_shufflevector(void)
+{
+#if defined(__GNUC__) && (defined(__clang__) || __GNUC__ >= 12)
```
Comment (kroening) [inline regression/cbmc/gcc_vector3/main.c]: This seems wrong, or at least hard to parse. Why would you check the GNUC version when using clang?

Comment (tautschnig) [reply regression/cbmc/gcc_vector3/main.c]: This is to restrict the GCC version: shufflevector is supported in all versions of clang, but for GCC it's only supported with version 12 or later. I can throw in an explicit `!defined(__clang__)` if that makes parsing easier?

Comment (kroening) [reply regression/cbmc/gcc_vector3/main.c]: How about
`#if defined(__clang__) || (defined(__clang__) && __GNUC__ >= 12)`

Comment (tautschnig) [reply regression/cbmc/gcc_vector3/main.c]: Done (except replacing the second use of `__clang__` by `__GNUC__`).

**Review by kroening — APPROVED** [review]
